### PR TITLE
Add Cursor.mogrify().

### PR DESCRIPTION
**Pull request by steveteahan**
Implements #476

### CODE REVIEW

File: MySQLdb/cursors.py [hunk 1]
```diff
@@ -182,6 +182,27 @@ def execute(self, query, args=None):
...
+
+        query -- string, query to execute on server
+        args -- optional sequence or mapping, parameters to use with query.
+
+        Note: If args is a sequence, then %s must be used as the
+        parameter placeholder in the query. If a mapping is used,
+        %(key)s must be used as the placeholder.
+
+        Returns bytes or bytearray representing the query
+        """
```
Comment (methane) [inline MySQLdb/cursors.py:205]: don't need long docstring for private function.

File: MySQLdb/cursors.py [hunk 2]
```diff
@@ -202,9 +223,22 @@ def execute(self, query, args=None):
...
+        query -- string, query to execute on server
+        args -- optional sequence or mapping, parameters to use with query.
+
+        Note: If args is a sequence, then %s must be used as the
+        parameter placeholder in the query. If a mapping is used,
+        %(key)s must be used as the placeholder.
+
+        Returns string representing query that would be executed by the server
+        """
+        return self._mogrify(query, args).decode()
```
Comment (methane) [inline MySQLdb/cursors.py:241]:
```suggestion
        return self._mogrify(query, args).decode(self._get_db().encoding)
```

File: MySQLdb/cursors.py [hunk 3]
```diff
@@ -202,9 +223,22 @@ def execute(self, query, args=None):
...
-        assert isinstance(query, (bytes, bytearray))
-        res = self._query(query)
-        return res
+        return query
+
+    def mogrify(self, query, args=None):
+        """Get the query exactly as it would be sent to the database running the
+        execute() method.
+
+        query -- string, query to execute on server
```
Comment (methane) [inline MySQLdb/cursors.py:232]: "execute on server" is wrong.

**Comment by steveteahan** [comment]
Thank you for the comments.  They have all been implemented.

**Comment by jmcopeland** [comment]
Came here looking for this... +1 for getting it merged.

**Pull request merged**